app/src/main/java/dmar/akan/akademiaandroida/features/characters/data/local/model/CharacterCached.kt

```diff
@@ -0,0 +1,49 @@
+package dmar.akan.akademiaandroida.features.characters.data.local.model
+
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+import dmar.akan.akademiaandroida.features.characters.domain.model.Character
+
+@Entity
+data class CharacterCached(
+    @PrimaryKey
+    val id: Int,
+    val name: String,
+    val species: String,
+    val type: String,
+    val gender: String,
+    val origin: Character.CharacterOrigin,
```

Comment:
W tym miejscu nie powinniśmy używać modelów z warstwy domeny, tylko ich odpowiedniki z warstwy danych. Już stworzyłeś taki model i jest to `CharacterOriginCached`. Wystarczy go tylko tutaj użyć. Dodatkowo zamiast tworzyć `TypeConverter` można użyć adnotacji `@Embeded`, która pozwoli na "rozpakowanie" obiektu i zapisaniu go w postaci typów prostych w bazie danych. 

```suggestion
    @Embeded("characterOriginCached")
    val origin: CharacterOriginCached,
```